juju/python-libjuju
# Cleanup release process for 3.x

#### Description

This clears up the release process for the master branch. Apparently I forgot to forward port some of the changes from  #921 (that clears it up for the 2.9 branch).

A twine check is added for validating the release package *before* trying to upload it.

The sphinx sections needed a fix for the rendering of the changelog. Specific rules for sections can be found in here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

#### QA Steps

The following command will do the first part of the release process where we generate a package with `setup.py`, and then check that package with `twine check`.

```
 $ rm -rf dist && python setup.py sdist && twine check dist/* && rm -rf dist
```

So this command should produce an output that says `PASSED`:

```sh
....
....
....
copying juju/client/proxy/factory.py -> juju-3.2.2.0/juju/client/proxy
copying juju/client/proxy/proxy.py -> juju-3.2.2.0/juju/client/proxy
copying juju/client/proxy/kubernetes/proxy.py -> juju-3.2.2.0/juju/client/proxy/kubernetes
Writing juju-3.2.2.0/setup.cfg
creating dist
Creating tar archive
removing 'juju-3.2.2.0' (and everything under it)
Checking dist/juju-3.2.2.0.tar.gz: PASSED
```
